pax-web-tomcat/src/main/java/org/ops4j/pax/web/service/tomcat/internal/EmbeddedTomcat.java

```diff
@@ -215,7 +215,8 @@ void configure(Configuration configuration) {
 			tomcatResource = getClass().getResource("/tomcat-server.xml");
 		}
 
-		File configurationFile = new File(configuration.getConfigurationDir(),
+		configurationDir = configuration.getConfigurationDir();
+		File configurationFile = new File(configurationDir,
 				SERVER_CONFIG_FILE_NAME);
```

Comment:
The configurationDir is actually the directory where the config files are located. In case of a fragment this is null. The code for actually reading the file is in TomcatServerWrapper (in the method getDefaultContextXml(), which actually does return an URI. However in case the file is not in a fragment, I need to pass the information about the directory where this file might be from EmbeddedTomcat (where the configuration is read) to TomcatServerWrapper.